src/jit-format/jit-format.cs

```diff
@@ -640,66 +642,122 @@ public static bool RunClangTidy(List<string> filenames, string compileCommands,
+                            Console.WriteLine("\tclang-tidy " + tidyFix + " -checks=-*," + checks + fixErrors + " -header-filter=src/jit/.* -p " + compileCommands + " " + filename);
                         }
-                        
-                        if (!fix)
+
+                        Process process = new Process();
+                    
+                        // Configure the process using the StartInfo properties.
+                        process.StartInfo.FileName = "clang-tidy";
+                        process.StartInfo.Arguments = tidyFix + "-checks=-*," + checks + fixErrors + " -header-filter=src/jit/.* -p " +
+                            compileCommands + " " + filename;
+                        process.StartInfo.UseShellExecute = false;
+                        process.StartInfo.RedirectStandardOutput = true;
+                        process.StartInfo.RedirectStandardError = true;
+                        process.StartInfo.CreateNoWindow = true;
+                        process.Start();
+
+                        string stdoutx = process.StandardOutput.ReadToEnd();         
+                        string stderrx = process.StandardError.ReadToEnd();             
+                        process.WaitForExit();
+
+
+                        if ((stdoutx.Contains("warning:") || (!ignoreErrors && stdoutx.Contains("error:"))))
                         {
-                            formatOk = false;
+                            if (verbose)
+                            {
+                                Console.WriteLine("clang-tidy: there are formatting errors in {0}", filename);
+                            }
                         }
-                    }
 
-                    if (!ignoreErrors && stderrx.Contains("error:"))
-                    {
-                        Console.Error.WriteLine("Error in clang-tidy: {0}", stderrx);
-                        
-                        if (!fix)
+                        if (!ignoreErrors && stderrx.Contains("error:"))
                         {
-                            formatOk = false;
+                            Console.Error.WriteLine("Error in clang-tidy: {0}", stderrx);
                         }
-                    }
-                        
-                    if (verbose)
-                    {
-                        if (stdoutx.Contains("warning:") || !ignoreErrors)
+                            
+                        if (verbose)
                         {
-                            Console.WriteLine(stdoutx);
+                            if (stdoutx.Contains("warning:") || !ignoreErrors)
+                            {
+                                Console.WriteLine(stdoutx);
+                            }
                         }
                     }
                 }
             }
+            else
+            {
+                Parallel.ForEach(filenames, (filename) =>
+                        {
```

Comment:
It's tough to tell just looking at this diff, but there appears to be a substantial amount of duplication between this loop and the one above that needs to be factored into helper methods.  You should either control the differences in behavior with method parameters or move as much as you can into one or more methods and leave the differences inside the loop bodies.  If that's not clear enough, let me know and I'll take a closer look.